### PR TITLE
libs: Update libdpkg from version v1.19.0.5 to v1.21.7

### PR DESCRIPTION
**Pull request by alessandrogario**
- Update the library `libdpkg`
- Make the `libdpkg` error handling more resilient
- Refactor of the `deb_packages` table
- Adds a column to the `deb_packages` table that makes the admin directory configurable instead of hardcoded to `/usr/local/var/lib/dpkg`

### CODE REVIEW

File: osquery/tables/system/linux/deb_packages.cpp [hunk 1]
```diff
@@ -7,196 +7,100 @@
...
+  auto error_code = error.getErrorCode();
 
-  if (pa->installed.arch == pb->installed.arch) {
-    return 0;
-  }
+  const auto& error_description =
+      IDpkgQuery::getErrorCodeDescription(error_code);
 
-  return strcmp(arch_a, arch_b);
+  logger.vlog(1,
```
Comment (directionless) [inline osquery/tables/system/linux/deb_packages.cpp:35]: If this is an actual error, do you think it belongs on verbose or info?

File: osquery/tables/system/linux/deb_packages.cpp [hunk 2]
```diff
@@ -7,196 +7,100 @@
...
+  } else {
+    admindir_list.push_back(kAdminDir);
   }
 
-  r["pid_with_namespace"] = "0";
-
-  results.push_back(r);
-}
+  auto dropper = DropPrivileges::get();
+  dropper->dropTo("nobody");
```
Comment (directionless) [inline osquery/tables/system/linux/deb_packages.cpp:56]: <3

**Review by directionless — APPROVED** [review]
I didn't test it, but the code seems fine

File: osquery/tables/system/linux/deb_packages.cpp (hunk 1 above)
Comment (alessandrogario) [reply osquery/tables/system/linux/deb_packages.cpp:35]: > If this is an actual error, do you think it belongs on verbose or info?

I think it probably makes sense to turn this into a real error. Thanks for spotting this!

**Review by directionless — APPROVED** [review]

**Pull request merged**